views/login.jsx

```diff
@@ -0,0 +1,215 @@
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          username: this.state.username,
+          password: this.state.password,
+          firstName: this.state.firstName,
+          lastName: this.state.lastName,
+          goldenTicket: this.state.goldenTicket,
+        }),
+      })
+      .then(response => response.json())
+      .then(jsonResponse => {
+        console.log(jsonResponse);
+        if (jsonResponse.message) {
+          this.setState({
+            ...this.state,
+            errorMessage: jsonResponse.message
+          });
+        }
+        else {
+          this.setState({
+            firstName: null,
+            lastName: null,
+            password: null,
+            goldenTicket: null,
+            registering: false,
+            successMessage: 'Account successfully created'
+          });
+        }
+      })
+      .catch(err => {
+        console.log(err);
+      });
+    }
+    else {
+      fetch('http://api.grebelife.com/users/login/', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          username: this.state.username,
+          password: this.state.password,
+        }),
+      })
+      .then(response => response.json())
+      .then(jsonResponse => {
+        console.log(jsonResponse);
+        if (jsonResponse.err) {
+          this.setState({
+            ...this.state,
+            errorMessage: jsonResponse.err.message
+          });
+        }
+        else {
+          this.props.navigation.navigate('Test');
+        }
+      })
+      .catch(err => {
+        console.log(err);
+      });
+    }
+  }
+
+  render() {
+    if (this.state.loading) {
+      return (
+        <Container>
+          <AppLoading/>
+        </Container>
+      );
+    }
+    else {
+      const registerFields = 
+        <View>
+          <Item floatingLabel>
+            <Label>First Name</Label>
+            <Input onChangeText={this.updateFirstName.bind(this)}/>
+          </Item>
+          <Item floatingLabel>
+            <Label>Last Name</Label>
+            <Input onChangeText={this.updateLastName.bind(this)}/>
+          </Item>
+          <Item floatingLabel>
+            <Label>Golden Ticket #</Label>
+            <Input onChangeText={this.updateGoldenTicket.bind(this)}/>
+          </Item>
+        </View>;
+
+      // Put these in own component
+      const errorBanner = 
+        <View style={{backgroundColor: '#C00', marginTop: 10}}>
+          <Text style={{paddingLeft: 10, color: 'white'}}>{this.state.errorMessage}</Text>
+        </View>;
+      const successBanner = 
+        <View style={{backgroundColor: '#0C0', marginTop: 10}}>
+          <Text style={{paddingLeft: 10, color: 'white'}}>{this.state.successMessage}</Text>
+        </View>;
```

Comment:
These banners should be moved into standalone components with a `message` prop or something